qcodes/data/hdf5_format.py

```diff
@@ -207,7 +211,12 @@ def write(self, data_set, io_manager=None, location=None,
             datasetshape = dset.shape
             old_dlen = datasetshape[0]
             x = data_set.arrays[array_id]
-            new_dlen = len(x[~np.isnan(x)])
+            try:
+                # get latest NaN element
+                new_dlen=(~np.isnan(x)).nonzero()[0][-1]+1
```

Comment:
could you "unwrap" this line of code so that it is clear what is going on? actually, what is going on? :) also, is it possible to avoid the use of `try-except` (if you are not "excepting" a very specific exception, this pattern can easily hide possible bugs in the code inside the `try` block)?